Automattic/pocket-casts-android
# Fixed tapping on the cast notification.

This change:
- Upgrades the CastOptionsProvider from Java to Kotlin
- Upgrades the [Cast library to 21.1.0](https://developers.google.com/cast/docs/release-notes#august-8,-2022)

Fixes https://github.com/Automattic/pocket-casts-android/issues/26

**Testing Steps**
1. Tap the 'Podcasts' page
2. Tap a podcast artwork to open a podcast page
3. Tap on the Chromecast icon in the toolbar 
4. Cast to a device
5. Tap play on an episode row
6. Tap the home button to exit the app
7. Pull down the notification shade
8. Tap the background of the notification

The Pocket Casts app should open again.
